packages/mask/src/plugins/Polls/Services.ts

```diff
@@ -101,59 +94,51 @@ export async function vote(props: voteProps) {
 
 export async function getPollByKey(props: { key: string }) {
     const keys = props.key.split('_')
-    let poll: PollGunDB = {
+
+    const node = await getGunData(PollGunRootNode, props.key)
+    if (typeof node !== 'object' || !isPollGunDB(node)) throw new TypeError('Invalid vote node.')
+    const data = node
+    const poll: PollGunDB = {
         ...defaultPoll,
-        key: props.key,
         id: first(keys),
+        key: props.key,
+        sender: data.sender,
+        question: data.question,
+        start_time: data.start_time,
+        end_time: data.end_time,
+    }
+    if (data.options) {
+        const options = await getGunData(PollGunRootNode, props.key, 'options')
```

Comment:
i think can provide a gundb agent object?

```
const gun = new GunAgent(/** prefix */ PollGunRootNode)

gun.get(...)
gun.set(...)
gun.delete(...)
// etc
```

Reply:
I explicitly limit the complexity of this wrapper package to make it easy to use.